app/frontend/src/components/UploadFile/UploadFile.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState, ChangeEvent, FormEvent } from "react";
+import styles from "./UploadFile.module.css";
+import { IIconProps, Callout, ActionButton, PrimaryButton, Label, IconButton, Text } from "@fluentui/react";
+import { SimpleAPIResponse, uploadFileApi, deleteUploadedFileApi, listUploadedFilesApi } from "../../api";
+import { useMsal } from "@azure/msal-react";
+import { useLogin, getToken } from "../../authConfig";
+
+interface Props {
+    className?: string;
+}
+
+export const UploadFile: React.FC<Props> = ({ className }: Props) => {
+    // State variables to manage the component behavior
+    const [isCalloutVisible, setIsCalloutVisible] = useState<boolean>(false);
+    const [isUploading, setIsUploading] = useState<boolean>(false);
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+    const [uploadedFile, setUploadedFile] = useState<SimpleAPIResponse>();
+    const [uploadedFiles, setUploadedFiles] = useState<string[]>([]);
+
+    if (!useLogin) {
+        console.error("This component requires useLogin to be true");
```

Comment:
should we raise an exception here? We would also have a conditional layout that never tries to render this component at all before we would hit this conditional correct?